spec/git-repository-async-spec.js

```diff
@@ -0,0 +1,598 @@
+      const workingDirPath = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirPath)
+      filePath = path.join(workingDirPath, 'a.txt')
+    })
+
+    it('no longer reports a path as modified after checkout', async () => {
+      let modified = await repo.isPathModified(filePath)
+      expect(modified).toBeFalsy()
+
+      fs.writeFileSync(filePath, 'ch ch changes')
+
+      modified = await repo.isPathModified(filePath)
+      expect(modified).toBeTruthy()
+
+      await repo.checkoutHead(filePath)
+
+      modified = await repo.isPathModified(filePath)
+      expect(modified).toBeFalsy()
+    })
+
+    it('restores the contents of the path to the original text', async () => {
+      fs.writeFileSync(filePath, 'ch ch changes')
+      await repo.checkoutHead(filePath)
+      expect(fs.readFileSync(filePath, 'utf8')).toBe('')
+    })
+
+    it('fires a did-change-status event if the checkout completes successfully', async () => {
+      fs.writeFileSync(filePath, 'ch ch changes')
+
+      await repo.getPathStatus(filePath)
+
+      const statusHandler = jasmine.createSpy('statusHandler')
+      repo.onDidChangeStatus(statusHandler)
+
+      await repo.checkoutHead(filePath)
+
+      expect(statusHandler.callCount).toBe(1)
+      expect(statusHandler.argsForCall[0][0]).toEqual({path: filePath, pathStatus: 0})
+
+      await repo.checkoutHead(filePath)
+      expect(statusHandler.callCount).toBe(1)
+    })
+  })
+
+  describe('.checkoutHeadForEditor(editor)', () => {
+    let filePath
+    let editor
+
+    beforeEach(() => {
+      spyOn(atom, "confirm")
+
+      const workingDirPath = copyRepository()
+      repo = new GitRepositoryAsync(workingDirPath, {project: atom.project, config: atom.config, confirm: atom.confirm})
+      filePath = path.join(workingDirPath, 'a.txt')
+      fs.writeFileSync(filePath, 'ch ch changes')
+
+      waitsForPromise(() => atom.workspace.open(filePath))
+      runs(() => editor = atom.workspace.getActiveTextEditor())
+    })
+
+    it('displays a confirmation dialog by default', async () => {
+      atom.confirm.andCallFake(({buttons}) => buttons.OK())
+      atom.config.set('editor.confirmCheckoutHeadRevision', true)
+
+      await repo.checkoutHeadForEditor(editor)
+
+      expect(fs.readFileSync(filePath, 'utf8')).toBe('')
+    })
+
+    it('does not display a dialog when confirmation is disabled', async () => {
+      atom.config.set('editor.confirmCheckoutHeadRevision', false)
+
+      await repo.checkoutHeadForEditor(editor)
+
+      expect(fs.readFileSync(filePath, 'utf8')).toBe('')
+      expect(atom.confirm).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('.destroy()', () => {
+    it('throws an exception when any method is called after it is called', async () => {
+      repo = new GitRepositoryAsync(require.resolve('./fixtures/git/master.git/HEAD'))
+      repo.destroy()
+
+      let threw = false
+      try {
+        await repo.getShortHead()
+      } catch (e) {
```

Comment:
Would it be better to catch this exception and assert the message and/or type? To ensure it throws the expected error and not just any error.